app/models/email_log.rb

```diff
@@ -40,6 +38,10 @@ class EmailLog < ActiveRecord::Base
     User.where(id: user_id).update_all("last_emailed_at = CURRENT_TIMESTAMP") if user_id.present?
   end
 
+  def topic
+    @topic ||= self.topic_id.present? ? Topic.find(self.topic_id) : self.post.topic
```

Comment:
We should be careful here, `Topic.find` will raise an error if `topic_id` is no longer valid.

Reply:
Ah right, I didn't think of the case where the topic is completely deleted, will use `find_by(id: self.topic_id)` instead. I am going to change this to `belongs_to :topic` in the next PR after I backfill anyway.